### ISSUE TITLE
Animate workspace switching as one continuous 1:1 animation

### ISSUE THREAD

**Issue by hanaral** [elementary/gala#1057]
## Problem

The 1:1 workspace switching only animates between the start workspace and the target workspace. The user has to lift their hand up and swipe again in the same direction if they want to see whether there are any more workspaces in that direction. This is also jarring for when you realise you are swiping to the wrong workspace and want to slide back and through the start workspace in to the next one over. 

## Proposal

The animation between and at the end of the workspace row should be a continuous 1:1 animation as if it was on a rail. MacOS and iOS do this exceptionally well.

## Prior Art

https://user-images.githubusercontent.com/58219504/108688815-3b40af00-74f0-11eb-9202-7550fb80f0eb.mp4

**Comment by hanaral** [comment]
@JoseExposito Here is the issue I mentioned before - since the workspace overview has got multiple rendered at once shouldn't it be able to just track the fingers like a Hdy carousel?

**Comment by JoseExposito** [comment]
Yes, it is technically possible, but it'll require some changes. Maybe something to implement in the future :+1:

**Comment by lenemter** [comment]
Fixed in https://github.com/elementary/gala/pull/2291
